### ISSUE TITLE
Updating global variable in class method makes the variable considered as class member. 

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#7948]
Currently pyright raise errors on following code by mistake
```python
_foo: int = 0 # Error: "_foo" is protected and used outside of the class in which it is declare

class Bar:
  def set_foo(self):
    global _foo
    _foo = 42
```

While following code passes check:
```python
_foo: int = 0

# non class member function
def set_foo():
  global _foo
  _foo = 42

# not updating 
class Bar:
  def print_foo(self):
    global _foo
    print(_foo)
```
Looks like updating a global variable in a class method make pyright thinks the variable is part of the class, even if it is not scoped with `self.` or `ClassName.`

I'm using pylance extension version `v2024.5.1`

**Comment by erictraut** [comment]
I'm not able to repro with the latest version of pyright. Please try with the prerelease version of pylance. I suspect that will fix it.

**Comment by yaxum62** [comment]
It's still reproducible from pre-release version `v2024.5.101 (pre-release)`
![image](https://github.com/microsoft/pyright/assets/8572778/94a82385-32fb-4c95-a417-215bcb7cfcf0)

**Comment by yaxum62** [comment]
Note I enabled strict mode, if that matters:
```
{
    "strict": ["**"],
}
```

**Comment by erictraut** [comment]
This was fixed in pyright 1.1.363. Pylance apparently hasn't yet picked up the fix. You should see the problem go away when the next prerelease version of pylance is published.
